il_nperf_cli.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+source autotest/user_input.cfg
+IFS='-' read -r -a cores <<< $core_list
+if [[ $1 == "-g" ]] || [[ $1 == "--generator" ]]; then
+	#s1u_cores="$(( ${cores[0]}+1 )).0"
+	s1u_cores="[$((${cores[0]}+1)):$((${cores[0]}+2))].0"
+	cmd="./app/x86_64-native-linuxapp-gcc/pktgen -l $core_list -n 4 --proc-type auto  \
+		--log-level 7 --socket-mem $NUMA0_MEM,$NUMA1_MEM --file-prefix s1u_pg -w $s1u_port -- -t -T     \
+		-N -m $s1u_cores -f ./autotest/input.txt -f themes/black-yellow.theme -g 127.0.0.1:2222"
+elif [[ $1 == "-r" ]] || [[ $1 == "--responder" ]]; then
+	#sgi_cores="[$(( ${cores[0]}+5 ))-$(( ${cores[0]}+6)):$(( ${cores[0]}+7 ))-$(( ${cores[0]}+8))].0"
+	sgi_cores="[$((${cores[0]}+3)):$((${cores[0]}+4))].0"
+	cmd="./app/x86_64-native-linuxapp-gcc/pktgen -l $core_list -n 4 --proc-type auto  \
+		--log-level 7 --socket-mem $NUMA0_MEM,$NUMA1_MEM --file-prefix sgi_pg -w $sgi_port -- -r -T     \
+		-N -m $sgi_cores -f ./autotest/input.txt -f themes/black-yellow.theme -g 127.0.0.1:3333"
```

Comment:
Asim : why do we have to put the -g flag even in responder mode (line#15 in il_nperf_cli.sh) ?

Reply:
This `-g` does not stand for "generator" mode. This switch is part of the original dpdk pktgen interface. Please see [this](https://pktgen-dpdk.readthedocs.io/en/latest/usage_pktgen.html) for more details. This switch is used to let the process create sockets and start listening on arbitrary TCP ports. The contents of the lua script (e.g. `start_il_trafficgen.lua`) are sent to these sockets (and, hence, can be used to control the operations of the processes).